### PR TITLE
Add default user to the syndesis-ci template

### PR DESCRIPTION
**Pull request by tplevko**
Created default test user for the syndesis-ci template. This is a really useful thing to have for our REST tests.

### CODE REVIEW

**Comment by dsimansk** [comment]
@rhuss fyi that "hack" I've mentioned yesterday. Related to our earlier conversations about system-tests.

[OT] might be better to change `probeless` annotation to something more explanatory like `ci`? (probes issue is gone anyway)

File: generator/01-header.yml.mustache [hunk 1]
```diff
@@ -12,7 +12,14 @@ parameters:
   required: true
 - name: GITHUB_OAUTH_CLIENT_SECRET
   description: GitHub OAuth client secret
+  required: true{{#Probeless}}
```
Comment (rhuss) [inline generator/01-header.yml.mustache:15]: I'd like to rename the 'Probeless' parameter to something like 'CI' as this matches better the purpose (as its not only about 'probeless', which btw is probably not necessary anymore to disable probes for CI as we are now on OpenShift 3.5. for the CI environment)

File: generator/01-header.yml.mustache [hunk 2]
```diff
@@ -12,7 +12,14 @@ parameters:
...
+
+- name: TEST_USER_GITHUB_USER_ID
+  description: Test user GitHub user ID
   required: true
+- name: TEST_USER_GITHUB_USER_NAME
```
Comment (rhuss) [inline generator/01-header.yml.mustache:20]: What's the difference between GITHUB_USER_ID and GITHUB_USER_NAME ?

File: generator/01-header.yml.mustache [hunk 3]
```diff
@@ -43,7 +50,21 @@ parameters:
...
+- name: TEST_USER_PASSWORD
+  description: The Keycloak test user password
+  generate: expression
+  from: "[a-zA-Z0-9]{10}"
+  required: true
+- name: TEST_USER_OPENSHIFT_USER_ID
+  description: OpenShift user id - default is ok for minishift, change it for OpenShift
+  value: 8f762a9b-7e71-11e7-868d-3eb6d1420952
+  required: true
+- name: TEST_USER_OPENSHIFT_USER_NAME
```
Comment (rhuss) [inline generator/01-header.yml.mustache:64]: Same here: what's the difference between ID and name ?

File: generator/03-syndesis-keycloak.yml.mustache [hunk 4]
```diff
@@ -900,7 +900,37 @@
...
           "xFrameOptions": "SAMEORIGIN",
           "contentSecurityPolicy": "frame-src 'self'"
-        },
+        },{{#Probeless}}
+
+        "users": [
+          {
+            "username": "test",
+            "enabled": true,
+            "email": "test@test.test",
```
Comment (rhuss) [inline generator/03-syndesis-keycloak.yml.mustache:909]: Would it make sense to add a real email address, just in case ? (not sure)

**Review by rhuss — COMMENTED** [review]
looks good, some questions though.

**Comment by rhuss** [comment]
@dsimansk is this PR still relevant ?

**Comment by tplevko** [comment]
@rhuss this can be rejected I guess. We have created another way for us to start with a default user. And what's more, the KC may no longer be our concern. As I understand it, it's future is not that certain with syndesis.

**Comment by rhuss** [comment]
Yes, KC will eventually vanish (hopefully soon). Going to close this PR then. Thanks !

**Pull request closed without merging**